apps/labs/posts/jupyter-everywhere.md

```diff
@@ -0,0 +1,234 @@
+
+<!-- add image of share dialog opened by clicking the "Share" button -->
+<!-- alt text: The share dialog in Jupyter Everywhere, opened by clicking the "Share" button in the toolbar, allowing users to generate view-only links for sharing their notebooks, and a "Copy link!" button to copy the generated link to the user's clipboard. -->
+
+There is some nuance to this: the view-only links are generated by hashing the notebook contents and session information, meaning that if the notebook is modified, the "Share" button needs to be interacted with again, or the user needs to wait until the next Jupyter auto-save occurs – which has a cadence of thirty seconds by default. This means that the shared link always points to the latest _saved_ version of the notebook, i.e, a snapshot of it. Users need to be aware that they may need to re-share the link if they make changes to the notebook after sharing it.
+
+Once the link is generated, users can share the link with others on any platform, such as email, messaging apps, or social media. Recipients can then click the link to view the notebook in their web browser, without installing any additional software or creating an account.
+
+When a user opens a shared notebook link, they are presented with a read-only view of the notebook, where they can navigate through the cells, view outputs, and interact with any visualisations or widgets embedded in the notebook. However, they cannot modify the notebook's contents or execute any cells.
+
+This was trickier to implement than we initially thought. At that time, JupyterLab did not expose a command for opening a notebook without a kernel attached. Fortunately, Jupyter is a swiss-army knife of extensibility, and we were able to override JupyterLab's default behaviour to allow opening a notebook in read-only mode without a kernel. This involved creating a custom notebook factory that would create a read-only notebook widget, and overriding the default kernel selection behaviour to prevent the user from selecting a kernel for the read-only notebook – by not instantiating a kernel at all.
+
+<!-- A section on the problem of non-persistence in Jupyter Everywhere due to no user accounts + how we addressed it -->
+
+#### Notebook downloads
+
+Another method for sharing notebooks is to export them from Jupyter Everywhere. This is facilitated by the built-in JupyterLite functionality to download notebooks in IPyNB format, which required adding a button to the notebook panel toolbar.
+
+However, we wanted to go a step further and allow users to download notebooks in PDF format as well. This is currently not supported out of the box in JupyterLite. JupyterLab relies on server-side components to generate PDFs from notebooks using `nbconvert` and LaTeX, creating high-quality PDFs suitable for printing and sharing. However, since Jupyter Everywhere runs entirely in the browser without any server-side components, this approach is not feasible.
```

Comment:
It might be nice to explain _why_ this isn't currently possible: `nbconvert` relies on `pandoc`, which isn't supported in the browser at the moment...